### PR TITLE
 enable incomplete/skipped/risky/warning for logger 

### PR DESCRIPTION
**Pull request by mario-naether**
- enable "logIncompleteSkipped" for JUnit logger
- enable config "beStrictAboutTestsThatDoNotTestAnything" and "beStrictAboutOutputDuringTests"

### CODE REVIEW

**Comment by edno** [comment]
I suggest to make `beStrictAboutTestsThatDoNotTestAnything` and `beStrictAboutOutputDuringTests ` configurable instead of hardcoding the value to **true**.
Also why not extending this feature to all parameters for [risky tests](https://phpunit.de/manual/current/en/risky-tests.html)?

**Comment by Naktibalda** [comment]
I think that it should be configurable with current behavior as default.
It isn't nice to change rules in the patch release.

**Comment by mario-naether** [comment]
I made the phpunit settings configurable in codeception settings and updated main yml file + documentation.

**Comment by edno** [comment]
You should default the parameters to **false** in [Codecept.php](https://github.com/Codeception/Codeception/blob/86770e89d266557c20dd0b1de5390e706f4770c1/src/Codeception/Codecept.php#L39), otherwise that will generate errors will running tests if the parameters are not set in `codeception.yml`.

**Comment by mario-naether** [comment]
ok, I added the default values as **false** in options array.

**Comment by mario-naether** [comment]
ok, I fixed the last stuff and added an method-check for old phpunit version

File: codeception.yml [hunk 1]
```diff
@@ -12,6 +12,10 @@ settings:
     memory_limit: 1024M
     log: true
     lint: true
+    reportUselessTests: false
```
Comment (DavertMik) [inline codeception.yml]: by convention and by other settings we use underscore_keys in yaml config files

File: docs/reference/Configuration.md [hunk 2]
```diff
@@ -63,6 +63,23 @@ settings:
...
+    # See https://phpunit.de/manual/current/en/risky-tests.html#risky-tests.useless-tests
+    reportUselessTests: false
+    
+    # PHPUnit can be strict about output during tests.
+    # See https://phpunit.de/manual/current/en/risky-tests.html#risky-tests.output-during-test-execution
+    disallowTestOutput: false
+    
+    # PHPUnit can be strict about tests that manipulate global state.
+    # See https://phpunit.de/manual/current/en/risky-tests.html#risky-tests.global-state-manipulation
+    beStrictAboutChangesToGlobalState: false
```
Comment (DavertMik) [inline docs/reference/Configuration.md]: tooooooooooooooooooooo long

what does it mean to be strict?
`no_global_changes`, `avoid_global_changes`,... 
naming can be improved

File: src/Codeception/Codecept.php [hunk 3]
```diff
@@ -37,29 +37,33 @@ class Codecept
...
+        'excludeGroups'                     => null,
+        'filter'                            => null,
+        'env'                               => null,
+        'fail-fast'                         => false,
+        'ansi'                              => true,
+        'verbosity'                         => 1,
+        'interactive'                       => true,
+        'no-rebuild'                        => false,
+        'quiet'                             => false,
+        'reportUselessTests'                => false,
```
Comment (DavertMik) [inline src/Codeception/Codecept.php]: this is not a configuration options but a runner options passed with parameters like `--fail-fast`, `--no-rebuild`. This is a config option so probably it should be set inside `Codeception\Configuration` class.

Yes, I know that might sound confusing :upside_down_face:

File: docs/reference/Configuration.md (hunk 2 above)
Comment (mario-naether) [inline docs/reference/Configuration.md]: this is a copy of the officially phpunit documentation

File: codeception.yml (hunk 1 above)
Comment (mario-naether) [inline codeception.yml]: ok, i changed the names settings to underscore.

File: src/Codeception/Codecept.php (hunk 3 above)
Comment (mario-naether) [inline src/Codeception/Codecept.php]: yes, i understand this and moved to Configuration.php

**Comment by DavertMik** [comment]
Yes, looks good. Thank you!

**Comment by Naktibalda** [comment]
@mario-naether @edno Would you miss `disallow_test_output`, `log_incomplete_skipped` and `report_useless_tests` settings if they were missing in Codeception 5?

**Comment by Naktibalda** [comment]
I reimplemented `report_useless_tests` and made it better than before: https://github.com/Codeception/Codeception/pull/6339

**Comment by edno** [comment]
> @mario-naether @edno Would you miss `disallow_test_output`, `log_incomplete_skipped` and `report_useless_tests` settings if they were missing in Codeception 5? [26b54ae](https://github.com/Codeception/Codeception/commit/26b54aec8c7cd652304e8d02147c9f7137dc2e0f)

I can work around if need be.

**Pull request merged**